recipes/liblifthttp/all/conanfile.py

```diff
@@ -0,0 +1,83 @@
+from conan import ConanFile
+from conan.tools.build import check_min_cppstd
+from conan.tools.cmake import CMake, CMakeDeps, CMakeToolchain, cmake_layout
+from conan.tools.files import copy, get
+
+import os
+
+required_conan_version = ">=2.0.9"
+
+class LiftHttpConan(ConanFile):
+    name = "lifthttp"
+    description = "Safe and easy to use C++17 HTTP client library."
+    license = "Apache-2.0"
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/jbaldwin/liblifthttp"
+    topics = ("cpp", "async", "asynchronous", "http", "https", "web", "client")
+    package_type = "library"
+    settings = "os", "arch", "compiler", "build_type"
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+    }
+    default_options = {
+        "shared": False,
+        "fPIC": True,
+    }
+
+    def export_sources(self):
+        copy(self, "conan_deps.cmake", self.recipe_folder, os.path.join(self.export_sources_folder, "src"))
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        if self.options.shared:
+            self.options.rm_safe("fPIC")
+
+        cmake_layout(self, src_folder="src")
+
+    def requirements(self):
+        self.requires("libcurl/[>=7.88.0 <9]", transitive_headers=True)
+        self.requires("libuv/[>=1.45.0 <2]", transitive_headers=True)
+
+    def source(self):
+        get(self, **self.conan_data["sources"][self.version], strip_root=True)
+
+    def validate(self):
+        check_min_cppstd(self, 17)
```

Comment:
```suggestion
        check_min_cppstd(self, 17)
        if is_msvc(self) and self.options.shared:
            # INFO: It lacks of exposing symbols in a .lib
            raise ConanInvalidConfiguration("The project does not support Windows shared build.")
```

@laci-aura I see you worked on [Windows support for liblifthttp](https://github.com/jbaldwin/liblifthttp/pull/161), but it still does not export those library symbols in a library, so people can consume it without using LoadLibrary() Windows API. I'll be opening an issue reporting it, but meanwhile let's avoid this configuration here.